apps/dot_ext/views/authorization.py

```diff
@@ -295,38 +296,43 @@ def dispatch(self, request, uuid, *args, **kwargs):
 class TokenView(DotTokenView):
     @method_decorator(sensitive_post_parameters("password"))
     def post(self, request, *args, **kwargs):
-        """Subclass of oauth2_provider.views.base.TokenView
-        to support checking application active/DAG status and
-        modification of the access token to include access
-        grant expiry.
-
-        RETURN:
-            response: HTTPResponse containing serialized access
-            token response.
-        """
+        dag = None
+        dag_expiry = None
         try:
-            dag = validate_app_is_active(request)[1]
-            if dag is not None and dag.expiration_date is not None:
-                dag_expiry = strftime('%Y-%m-%d %H:%M:%S', dag.expiration_date.timetuple())
-            else:
-                dag_expiry = None
+            app = validate_app_is_active(request)
         except (InvalidClientError, InvalidGrantError) as error:
             return json_response_from_oauth2_error(error)
 
+        if app.data_access_type == "THIRTEEN_MONTH":
+            try:
+                dag = DataAccessGrant.objects.get(
+                    beneficiary=request.user.id,
+                    application=app
+                )
+                if dag is not None and dag.expiration_date is not None:
```

Comment:
Any insight on what determines whether or not the dag returned has an expiration date? It's minor, but if we're recalculating it at all rather than pulling the dag value, we're likely to be off by at least a few seconds from what's actually in the db.